src/trace.zig

```diff
@@ -284,67 +227,73 @@ pub const Tracer = struct {
             "\"ph\":\"{[event]c}\"," ++
             "\"ts\":{[timestamp]}," ++
             "\"cat\":\"{[category]s}\"," ++
-            "\"name\":\"{[name]s}{[data]}\"," ++
+            "\"name\":\"{[category]s} {[event_tracing]} {[event_timing]}\"," ++
             "\"args\":{[args]s}" ++
             "}},\n", .{
             .process_id = tracer.replica_index,
-            .thread_id = event.stack(),
+            .thread_id = event_tracing.stack(),
             .category = @tagName(event),
             .event = 'B',
-            .timestamp = time_elapsed_us,
-            .name = event,
-            .data = struct_format(data, .sparse),
-            .args = if (comptime @TypeOf(data) == @TypeOf(.{}))
-                "{}" // Serialize `.{}` as an empty object, not as an empty array.
-            else
-                std.json.Formatter(@TypeOf(data)){ .value = data, .options = .{} },
+            .timestamp = time_elapsed.microseconds(),
+            .event_tracing = event_tracing,
+            .event_timing = event_timing,
+            .args = std.json.Formatter(Event){ .value = event, .options = .{} },
         }) catch unreachable;
```

Comment:
FWIW, I am slightly uneasy about this unreachable. We either should test that all events fit into the buffer, or maybe just `log.err` here instead of outright unreachable? Would be a pity if metrics kill us if we reach over 9000 ops or something, and it's not immediately obvious from the code that we are not overflowing this buffer here.